slice-core/src/main/java/com/cognifide/slice/core/internal/provider/SliceModelProvider.java

```diff
@@ -229,17 +234,21 @@ public Object get(String className, Resource resource) throws ClassNotFoundExcep
 	}
 
 	private <T> T get(Key<T> key, ExecutionContextImpl executionItem) {
+		InjectionMonitoringContext monitoringContext = sliceStats.startMonitoring();
+		monitoringContext.setInjecteeClass(key.getTypeLiteral().getRawType());
 		final ContextProvider oldContextProvider = contextScope.getContextProvider();
 		contextScope.setContextProvider(contextProvider);
 
 		if ((executionItem.getResource() == null) && (executionItem.getPath() != null)) {
 			executionItem.setPath(currentExecutionContext.getAbsolutePath(executionItem.getPath()));
 		}
-
+		executionItem.setInjecteeClass(key.getTypeLiteral().getRawType());
```

Comment:
See:
monitoringContext.setInjecteeClass(key.getTypeLiteral().getRawType());
vs.
executionItem.setInjecteeClass(key.getTypeLiteral().getRawType());

Do we need both lines?